BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Internal/Tracer.cs

```diff
@@ -102,7 +117,26 @@ public void OnSpanEnd(Span span)
                 }
                 return;
             }
-            Sample(span);
+            else
+            {
+                RunOnEndCallbacks(span);
+            }
+            if (!span.WasAborted)
+            {
+                Sample(span);
+            }
+        }
+
+        public void RunOnEndCallbacks(Span span)
+        {
+            foreach (var callback in _onSpanEndCallbacks)
+            {
+                if (!callback.Invoke(span))
```

Comment:
We should guard against the callback throwing an exception – catch it, log an error and then continue with the next callback.